### PR TITLE
KOGITO-5015: Workflows for publishing the Editors on NPM

### PR DESCRIPTION
**Pull request by caponetto**
Adding two new workflows:
- `publish_release_artifacts_npm`: This workflow is triggered after a release is done. It builds the repository and publishes the editors on NPM using the `tag` as the NPM version.

- `publish_snapshot_artifacts_npm`: This workflow is manually triggered. It builds the repository and publishes the editors on NPM using the `$MVN_VERSION-$DATE.$TIME` format as the NPM version.

### CODE REVIEW

File: .github/workflows/publish_snapshot_artifacts_npm.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          ls -la .
+          npm version ${{ steps.editors_version_npm.outputs.version }}
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.KIEGROUP_NPM_TOKEN }}" > ~/.npmrc
+          npm publish --access public --tag snapshot
+
+      - name: Publish SceSim Editor SNAPSHOT
+        run: |
+          cd ${{ github.workspace }}/.github/supporting-files/publish-artifacts-npm/scesim-editor-unpacked
+          cp -r ${{ github.workspace }}/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/target/drools-wb-scenario-simulation-editor-kogito-runtime ./target/scesim
+          ls -la .
```
Comment (tiagobento) [inline .github/workflows/publish_snapshot_artifacts_npm.yml:53]: Leftover? :D

File: .github/supporting-files/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+!target/
```
Comment (tiagobento) [inline .github/supporting-files/.gitignore:1]: Can't remember why this is here, to be honest. Can you please explain it?

File: .github/workflows/publish_release_artifacts_npm.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          ls -la .
+          npm version ${{ steps.release_tag.outputs.tag }}
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.KIEGROUP_NPM_TOKEN }}" > ~/.npmrc
+          npm publish --access public
+
+      - name: Publish SceSim Editor
+        run: |
+          cd ${{ github.workspace }}/.github/supporting-files/publish-artifacts-npm/scesim-editor-unpacked
+          cp -r ${{ github.workspace }}/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/target/drools-wb-scenario-simulation-editor-kogito-runtime ./target/scesim
+          ls -la .
```
Comment (tiagobento) [inline .github/workflows/publish_release_artifacts_npm.yml:50]: Leftover?

**Comment by tiagobento** [comment]
@caponetto Remember to add the NPM token secret before merging this PR.

File: .github/supporting-files/.gitignore (hunk 2 above)
Comment (caponetto) [reply .github/supporting-files/.gitignore:1]: It came from https://github.com/kiegroup/kogito-tooling/pull/401.

File: .github/workflows/publish_snapshot_artifacts_npm.yml (hunk 1 above)
Comment (caponetto) [reply .github/workflows/publish_snapshot_artifacts_npm.yml]: Removed in https://github.com/kiegroup/kogito-editors-java/pull/10/commits/e1d415c9b20289832eca6a5380fc0be2d9ae0377.

File: .github/workflows/publish_release_artifacts_npm.yml (hunk 3 above)
Comment (caponetto) [reply .github/workflows/publish_release_artifacts_npm.yml]: Removed in https://github.com/kiegroup/kogito-editors-java/pull/10/commits/e1d415c9b20289832eca6a5380fc0be2d9ae0377.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**